### PR TITLE
refactor: pipeline custom command line support for newlines

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
pipeline custom command line support for newlines

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?id=325979&iterationID=1337&pId=0&type=REQUIREMENT)
- [custom shell link](https://erda.cloud/erda/dop/projects/387/issues/gantt?id=328190&iterationID=1337&pId=0&type=REQUIREMENT)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |   pipeline custom command line support for newlines           |
| 🇨🇳 中文    |   pipeline自定义命令支持换行           |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: internal/tools/pipeline/actionagent/step_prepare_test.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+                        --proxy \
+                        --env-default mall-((PROJECT_TYPE))/.env.default`,
+			want: `#!/bin/sh
+set -e
+
+echo + "npm i @terminus/trnw-cli -g\n            trnw-cli build apk -T -A -t assembleRelease \\\n                        --proxy \\\n                        --env-default mall-((PROJECT_TYPE))/.env.default"
+npm i @terminus/trnw-cli -g
+            trnw-cli build apk -T -A -t assembleRelease \
+                        --proxy \
+                        --env-default mall-((PROJECT_TYPE))/.env.default || ((echo "- FAIL! exit code: $?") && false)
```
Comment (sfwn) [inline internal/tools/pipeline/actionagent/step_prepare_test.go:58]: I think block script should be different with multi-line commands.
We should let the use know which concrete line in block script fails.

File: internal/tools/pipeline/actionagent/step_prepare_test.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+echo + "1"
+1 || ((echo "- FAIL! exit code: $?") && false)
+echo
+
+`,
+		},
+		{
+			name:     "invalid commands",
+			commands: 123,
```
Comment (sfwn) [inline internal/tools/pipeline/actionagent/step_prepare_test.go:89]: I think use can input 123.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: internal/tools/pipeline/actionagent/step_prepare_test.go (hunk 1 above)
Comment (chengjoey) [reply internal/tools/pipeline/actionagent/step_prepare_test.go:58]: if the user wrote block script, it's difficult to split each command
for instance:
```
git clone xxx \
-depth=1 \
-v
```

```
cat /tmp/xxx
sleep 10
```

File: internal/tools/pipeline/actionagent/step_prepare_test.go (hunk 2 above)
Comment (chengjoey) [reply internal/tools/pipeline/actionagent/step_prepare_test.go:89]: adjusted

File: internal/tools/pipeline/actionagent/step_logic.go [hunk 3]
```diff
@@ -47,3 +58,24 @@ func (agent *Agent) logic() {
...
+	commandRun.Stdout = io.MultiWriter(agent.EasyUse.RunMultiStdout)
+	commandRun.Stderr = io.MultiWriter(agent.EasyUse.RunMultiStderr)
+	if err := commandRun.Start(); err != nil {
+		agent.AppendError(err)
+		return
+	}
+	agent.EasyUse.RunProcess = commandRun.Process
+	if err := commandRun.Wait(); err != nil {
+		agent.ExitCode = 1
+		if err.Error() != fmt.Sprintf(`command "%s": exit status 1`, agent.EasyUse.CommandScript) {
```
Comment (sfwn) [inline internal/tools/pipeline/actionagent/step_logic.go:76]: you should get the truly exit code

**Review by sfwn — CHANGES_REQUESTED** [review]

File: internal/tools/pipeline/actionagent/step_logic.go (hunk 3 above)
Comment (chengjoey) [reply internal/tools/pipeline/actionagent/step_logic.go]: adjusted

**Review by sfwn — APPROVED** [review]

**Pull request merged**